torch/csrc/jit/passes/frozen_concat_linear.cpp

```diff
@@ -0,0 +1,186 @@
+
+        // Now iterate over the rest of the users of the set to
+        // see if there is anything that we can coaleasce with.
+        for (auto it = uses.begin(); it != uses.end();) {
+          auto candidate_node = *it;
+          auto candidate_weight = constant_as<Tensor>(candidate_node->namedInput("weight"));
+          auto candidate_weight_type = candidate_weight->dtype();
+          auto candidate_bias = constant_as<Tensor>(candidate_node->namedInput("bias"));
+          auto candidate_bias_type = candidate_bias->dtype();
+
+          // For now we will just keep it simple and require matching types
+          // Type promotion mgiht cause performance to actually decrease.
+          if (weight_type != candidate_weight_type || bias_type != candidate_bias_type) {          
+            it++;
+            continue;
+          }
+
+          bool can_move_before_all = true;
+          for (auto n : matching_nodes) {
+            can_move_before_all &=
+                getAliasDb()->couldMoveBeforeTopologically(candidate_node, n);
+          }
+          if (!can_move_before_all) {
+            it++;
+            continue;
+          }
+
+          // Found a node that is eligible for combination
+          auto prev = it;
+          it++;
+          uses.erase(prev);
+          matching_nodes.push_back(candidate_node);
+        }
+        if (matching_nodes.size() == 1) {
+          // Found no other compatible linear layers, do nothing.
+          continue;
+        }
+        graph_modified = true;
+
+        for (auto n : matching_nodes) {
+          if (n == base_node) {
+            continue;
+          }
+          getAliasDb()->moveBeforeTopologicallyValid(base_node, n);
+        }
+
+        // Now actually merge the candidate nodes
+        graph_->setInsertPoint(base_node);
+        Value* zero = graph_->insertConstant(0);
+
+        auto weight_list = c10::fmap(
+            matching_nodes, [](Node* n) { return n->namedInput("weight"); });
+        auto weight_with_dim(weight_list);
+        weight_with_dim.push_back(zero);
+
+        auto bias_with_dim = c10::fmap(
+            matching_nodes, [](Node* n) { return n->namedInput("bias"); });
+        bias_with_dim.push_back(zero);
+
+        auto weight_cat = graph_->create(prim::VarConcat, weight_with_dim);
+        weight_cat->insertBefore(base_node);
+        auto bias_cat = graph_->create(prim::VarConcat, bias_with_dim);
+        bias_cat->insertBefore(base_node);
+
+        // create the new node
+        std::vector<Value*> linear_in = {
+            op_value, weight_cat->output(), bias_cat->output()};
+        auto linear_node = graph_->create(aten::linear, linear_in);
+        linear_node->insertBefore(base_node);
+
+        // Update the outputs of the nodes
+        graph_->setInsertPoint(linear_node);
+        Value* neg1 = graph_->insertConstant(-1);
+        Value* one = graph_->insertConstant(1);
+
+        long long cur_loc = 0;
```

Comment:
i dont think i've seen `long long` used in pytorch code base at all.. `size_t` ?

Reply:
is `size_t` or `int64_t` a better choice here. Also looks like due to `long long` not being a standard size, I am getting compiler errors on some targets.

Reply:
either is fine probably. size_t i guess since this is always positive